### ISSUE TITLE
Large memory consumption with --fastx_revcomp

### ISSUE THREAD

**Issue by frederic-mahe** [torognes/vsearch#111]
Hi Torbjørn,

I am testing the new `--fastx_revcomp` function and I observe a very large memory consumption (97% of 32 GB of RAM) when reverse-complementing a 7 GB fastq file.
```
Tasks: 509 total,   3 running, 506 sleeping,   0 stopped,   0 zombie
%Cpu(s):  0,2 us,  3,0 sy,  0,0 ni, 90,5 id,  6,3 wa,  0,0 hi,  0,0 si,  0,0 st
KiB Mem:  32901960 total, 32645668 used,   256292 free,     1420 buffers
KiB Swap: 33510396 total, 12835796 used, 20674600 free.    12552 cached Mem

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                                                                                                                       
22864 lstm      20   0 84,754g 0,030t    212 R  67,8 97,4   5:11.97 vsearch 
```
Is it normal? if not, can you reproduce that bug?

**Comment by torognes** [comment]
Looks like a memory leak. Will check.

**Comment by frederic-mahe** [comment]
Yes. A small datasets generates an error with valgrind
```
valgrind --leak-check=full vsearch --fastx_revcomp test.fastq --fastqout test_rev.fastq 
==24125== Memcheck, a memory error detector
==24125== Copyright (C) 2002-2013, and GNU GPL'd, by Julian Seward et al.
==24125== Using Valgrind-3.10.1 and LibVEX; rerun with -h for copyright info
==24125== Command: vsearch --fastx_revcomp test.fastq --fastqout test_rev.fastq
==24125== 
vsearch v1.3.0_linux_x86_64, 31.4GB RAM, 32 cores
https://github.com/torognes/vsearch

Reading fastq file 100%  
==24125== 
==24125== HEAP SUMMARY:
==24125==     in use at exit: 821,432 bytes in 101 blocks
==24125==   total heap usage: 413 allocs, 312 frees, 1,011,606 bytes allocated
==24125== 
==24125== 2,232 bytes in 1 blocks are definitely lost in loss record 1 of 2
==24125==    at 0x4C2E136: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==24125==    by 0x4C2E251: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==24125==    by 0x420553: xmalloc(unsigned long) (util.cc:146)
==24125==    by 0x412AB2: fastq_open(char const*) (fastq.cc:153)
==24125==    by 0x415BC6: fastx_revcomp() (fastqops.cc:803)
==24125==    by 0x4022F7: main (vsearch.cc:2027)
==24125== 
==24125== 819,200 bytes in 100 blocks are definitely lost in loss record 2 of 2
==24125==    at 0x4C2E136: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==24125==    by 0x4C2E251: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==24125==    by 0x420553: xmalloc(unsigned long) (util.cc:146)
==24125==    by 0x4136CA: buffer_init (fastq.cc:105)
==24125==    by 0x4136CA: fastq_next(fastq_s*, bool, char*) (fastq.cc:540)
==24125==    by 0x415C6E: fastx_revcomp() (fastqops.cc:825)
==24125==    by 0x4022F7: main (vsearch.cc:2027)
==24125== 
==24125== LEAK SUMMARY:
==24125==    definitely lost: 821,432 bytes in 101 blocks
==24125==    indirectly lost: 0 bytes in 0 blocks
==24125==      possibly lost: 0 bytes in 0 blocks
==24125==    still reachable: 0 bytes in 0 blocks
==24125==         suppressed: 0 bytes in 0 blocks
==24125== 
==24125== For counts of detected and suppressed errors, rerun with: -v
==24125== ERROR SUMMARY: 2 errors from 2 contexts (suppressed: 0 from 0)
```

**Comment by torognes** [comment]
Fixed now in 1.3.2, together with a few other memory leaks.
